pkg/controllers/management/clusterprovisioner/provisioner.go

```diff
@@ -89,7 +89,13 @@ func (p *Provisioner) Remove(cluster *v3.Cluster) (runtime.Object, error) {
 			break
 		}
 		if i == 3 {
-			return cluster, fmt.Errorf("failed to remove the cluster [%s]: %v", cluster.Name, err)
+			if strings.Contains(err.Error(), "not found") {
```

Comment:
Figure out what the error is, possibly type of error to ensure we only do this when expected.